bokehjs/src/lib/models/glyphs/webgl/marker.frag

```diff
@@ -212,6 +239,130 @@ float marker_distance(in vec2 p, in int line_cap, in int line_join)
 }
 #endif
 
+#if defined(USE_ANNULUS) || defined(USE_WEDGE) || defined(USE_ANNULAR_WEDGE)
+float merge(in float d1, in float d2)
+{
+  return min(d1, d2);
+}
+
+float intersect(in float d1, in float d2)
+{
+  return max(d1, d2);
+}
+
+float subtract(in float d1, in float d2)
+{
+  return max(d1, -d2);
+}
+
+float cro2(in vec2 a, in vec2 b)
+{
+  return a.x*b.y - a.y*b.x;
+}
+
+float circle(in vec2 p, in float radius)
+{
+  return length(p) - radius;
+}
+
+float segment_square(in vec2 p, in vec2 q) {
+  vec2 v = p - q*clamp(dot(p, q)/dot(q, q), 0.0, 1.0);
+  return dot(v, v);
+}
+
+vec2 xy(in float angle)
+{
+  return vec2(cos(angle), sin(angle));
+}
+
+// From https://www.shadertoy.com/view/wldXWB (MIT licensed)
+float wedge(in vec2 p, in float r, in float start_angle, in float end_angle)
+{
+    vec2 a = r*xy(start_angle);
+    vec2 b = r*xy(end_angle);
+
+    // distance
+    float d = sqrt(merge(segment_square(p, a), segment_square(p, b)));
+
+    // sign
+    float s;
+    if (cro2(a, b) < 0.0) {
+        s =  sign(max(cro2(a, p), cro2(p, b)));
+    } else {
+        s = -sign(max(cro2(p, a), cro2(b, p)));
+    }
+
+    return s*d;
+}
+
+float annulus(in vec2 p, in float outer_radius, in float inner_radius)
+{
+  float outer = circle(p, outer_radius);
+  float inner = circle(p, inner_radius);
+
+  return subtract(outer, inner);
+}
+#endif
+
+#if defined(USE_ANNULUS)
+float marker_distance(in vec2 p, in int line_cap, in int line_join)
+{
+  return annulus(p, v_outer_radius, v_inner_radius);
+}
+#endif
+
+#if defined(USE_WEDGE)
+float marker_distance(in vec2 p, in int line_cap, in int line_join)
+{
+  return intersect(
+    circle(p, v_radius),
+    wedge(p, v_radius, v_start_angle, v_end_angle));
+}
+#endif
+
+#if defined(USE_ANNULAR_WEDGE)
+float marker_distance(in vec2 p, in int line_cap, in int line_join)
+{
+  return intersect(
+    annulus(p, v_outer_radius, v_inner_radius),
+    wedge(p, v_outer_radius, v_start_angle, v_end_angle));
+}
+#endif
+
+#if defined(USE_ELLIPSE)
+// From https://www.shadertoy.com/view/tttfzr (MIT licensed)
```

Comment:
I am not really comfortable with copying someone else's code and pasting it into Bokeh. Ideally I would like the person submitting such code to demonstrate they understand it and take ownership of it in terms of supporting it in future, for two reasons:
1. I'd like to ask about the iterative Newton solver in the code block below. Why iterative? Why 5 iterations? Does it cope well with long thin ellipses? Is the `0.01` value critical or would `0.0` be OK?
2. Eventually the code for some of these new shapes is going to have to support line joins and line dashes, so it is going to need to be understood and modified.

Reply:
@ianthomas23 it looks like this ellipse code has been removed for now, are you otherwise happy with this PR?

Reply:
I'm pretty sure I commented earlier, but maybe I didn't click "Comment". In anycase, I removed the implementation of ellipse glyph, because it's the most complex of new glyphs, one that I don't fully understand and it's not urgently needed anyway. I may get back to this at some point, when I can actually explain the details of this implementation.